include/swift/Driver/Action.h

```diff
@@ -144,6 +146,30 @@ class CompileJobAction : public JobAction {
     static InputInfo makeNewlyAdded() {
       return InputInfo(Status::NewlyAdded, llvm::sys::TimeValue::MaxTime());
     }
+
+    /// \returns The string identifier used to represent the given status in a
+    /// compilation record file (.swiftdeps file).
+    static StringRef getStatusName(Status S) {
```

Comment:
Since this doesn't round-trip cleanly, I think we should give it a different name.